pallets/connectors/src/inbound.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_traits::{
+	ops::{EnsureAdd, EnsureSub},
+	CurrencyInspect, Investment, InvestmentCollector, Permissions,
+};
+use cfg_types::{
+	domain_address::{Domain, DomainAddress},
+	permissions::{PermissionScope, PoolRole, Role},
+};
+use frame_support::{
+	ensure,
+	traits::fungibles::{Mutate, Transfer},
+};
+use sp_runtime::{
+	traits::{Convert, Zero},
+	DispatchResult,
+};
+
+use crate::{
+	pallet::Error, Config, CurrencyIdOf, GeneralCurrencyIndexOf, Pallet, PoolIdOf, TrancheIdOf,
+};
+
+impl<T: Config> Pallet<T> {
+	/// Executes a transfer from another domain exclusively for
+	/// non-tranche-tokens.
+	///
+	/// Directly mints the currency into the receiver address.
+	pub fn do_transfer_from_other_domain(
```

Comment:
```suggestion
	pub fn do_transfer(
```

or 

```suggestion
	pub fn do_transfer(
```


Given that this module is called `inbound`, the context of these functions being about messages coming from other domains is a given, so I would drop the redundant and verbose `_from_other_domain(`. Using `handle` instead of `do` also suggests that it's handling something that's coming in, but maybe that's just how I interpret that.